### PR TITLE
[13.0][MIG] website_form_recaptcha

### PR DESCRIPTION
**Pull request by bjornbillen**

### CODE REVIEW

File: website_form_recaptcha/models/website_form_recaptcha.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                "The response parameter is invalid or malformed."
+            ),
+        }
+        return mapping.get(
+            errorcode, _("There was a problem with " "the captcha entry.")
+        )
+
+    def _get_api_credentials(self, website=None):
+        # website override
+        website = website or http.request.website
```
Comment (tarteo) [inline website_form_recaptcha/models/website_form_recaptcha.py:43]: System parameter `recaptcha.key.site` and `recaptcha.key.secret` do nothing.

**Review by tarteo — CHANGES_REQUESTED** [review]
Tested, looks good to me except one remark.

File: website_form_recaptcha/models/website_form_recaptcha.py (hunk 1 above)
Comment (bjornbillen) [reply website_form_recaptcha/models/website_form_recaptcha.py:43]: @tarteo : Can you elaborate a little more on this? I can't see any config parameters being created by the module? There's something written in the module description that refers to the system parameters, which should probably be removed...

Comment (simahawk) [reply website_form_recaptcha/models/website_form_recaptcha.py:43]: @tarteo ping

File: website_form_recaptcha/models/website_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    has_google_recaptcha = fields.Boolean(
+        "Google reCaptcha",
+        compute="_compute_has_google_recaptcha",
+        inverse="_inverse_has_google_recaptcha",
+        readonly=False,
+    )
+
+    @api.depends("website_id")
+    def _compute_has_google_recaptcha(self):
+        self.has_google_recaptcha = bool(self.recaptcha_key_site)
```
Comment (bizzappdev) [inline website_form_recaptcha/models/website_config_settings.py]: here there should be a for loop for config in self:

File: website_form_recaptcha/models/website_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    def _inverse_has_google_recaptcha(self):
```
Comment (bizzappdev) [inline website_form_recaptcha/models/website_config_settings.py:28]: here there should be a for loop for config in self:

File: website_form_recaptcha/models/website_form_recaptcha.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            "missing-input-response": _("The response parameter is missing."),
+            "invalid-input-response": _(
+                "The response parameter is invalid or malformed."
+            ),
+        }
+        return mapping.get(
+            errorcode, _("There was a problem with " "the captcha entry.")
+        )
+
+    def _get_api_credentials(self, website=None):
```
Comment (bizzappdev) [inline website_form_recaptcha/models/website_form_recaptcha.py]: IMO @api.model is needed for this function.

**Review by bizzappdev — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
ping @bjornbillen please resolve conflicts

**Comment by Tardo** [comment]
Well, i saw that @bjornbillen is no longer active on github... so i will supersede this pr.

**Comment by bjornbillen** [comment]
@tardo Go ahead mate. IMO, it takes way too long to get changes done over here. Currently can't affort to waste such time anymore.

**Comment by Tardo** [comment]
Oh! @bjornbillen sorry xD i have not seen any activity on your feed :P

**Comment by pedrobaeza** [comment]
Superseded by #709

**Pull request closed without merging**